test/uk/gov/hmrc/brm/metrics/MetricsSpec/MetricsSpec.scala

```diff
@@ -0,0 +1,224 @@
+    }
+
+  }
+
+  "NRSMetrics" should {
+
+    "initialise" in {
+      val metrics = NRSMetrics
+      metrics shouldBe a [Metrics]
+    }
+
+    "have a timer for the nrs connection" in {
+      val metrics = NRSMetrics
+      metrics.time(System.currentTimeMillis(), TimeUnit.MILLISECONDS)
+      MetricsRegistry.defaultRegistry.getTimers.get("nrs-timer").getCount shouldBe 1
+    }
+
+    "have a 200 status count for nrs" in {
+      val metrics = NRSMetrics
+      metrics.connectorStatus(200)
+      MetricsRegistry.defaultRegistry.getCounters.get("nrs-connector-status-200").getCount shouldBe 1
+    }
+
+    "have a 400 status count for nrs" in {
+      val metrics = NRSMetrics
+      metrics.connectorStatus(400)
+      MetricsRegistry.defaultRegistry.getCounters.get("nrs-connector-status-400").getCount shouldBe 1
+    }
+
+    "have a 404 status count for proxy" in {
+      val metrics = NRSMetrics
+      metrics.connectorStatus(404)
+      MetricsRegistry.defaultRegistry.getCounters.get("nrs-connector-status-404").getCount shouldBe 1
+    }
+
+    "have a 500 status count for nrs" in {
+      val metrics = NRSMetrics
+      metrics.connectorStatus(500)
+      MetricsRegistry.defaultRegistry.getCounters.get("nrs-connector-status-500").getCount shouldBe 1
+    }
+
+    "have a 502 status count for proxy" in {
+      val metrics = NRSMetrics
+      metrics.connectorStatus(502)
+      MetricsRegistry.defaultRegistry.getCounters.get("nrs-connector-status-502").getCount shouldBe 1
+    }
+
+    "have a 504 status count for proxy" in {
+      val metrics = NRSMetrics
+      metrics.connectorStatus(504)
+      MetricsRegistry.defaultRegistry.getCounters.get("nrs-connector-status-504").getCount shouldBe 1
+    }
+
+    "accept a status code not registered" in {
+      val metrics = NRSMetrics
+      for (i <- 1 to 5) yield metrics.connectorStatus(423)
+      MetricsRegistry.defaultRegistry.getCounters.get("nrs-connector-status-423").getCount shouldBe 5
+    }
+
+  }
+
+  "GRONIMetrics" should {
+
+    "initialise" in {
+      val metrics = GRONIMetrics
+      metrics shouldBe a [Metrics]
+    }
+
+    "have a timer for the gro-ni connection" in {
+      val metrics = GRONIMetrics
+      metrics.time(System.currentTimeMillis(), TimeUnit.MILLISECONDS)
+      MetricsRegistry.defaultRegistry.getTimers.get("gro-ni-timer").getCount shouldBe 1
+    }
+
+    "have a 200 status count for gro-ni" in {
+      val metrics = GRONIMetrics
+      metrics.connectorStatus(200)
+      MetricsRegistry.defaultRegistry.getCounters.get("gro-ni-connector-status-200").getCount shouldBe 1
+    }
+
+    "have a 400 status count for gro-ni" in {
+      val metrics = GRONIMetrics
+      metrics.connectorStatus(400)
+      MetricsRegistry.defaultRegistry.getCounters.get("gro-ni-connector-status-400").getCount shouldBe 1
+    }
+
+    "have a 404 status count for proxy" in {
+      val metrics = GRONIMetrics
+      metrics.connectorStatus(404)
```

Comment:
Ah, so we are a little more specific with error codes. Do you want us to log errors outside of 200, 400 and 500 (if applicable)